gitpod-io/gitpod
# Use new cron plugin for Werft cron jobs

## Description

This moves our existing Werft cron jobs to use the cron plugin which was released in Werft 0.2.0. This greatly simplifies the process for adding and updating Werft cron jobs as you no longer have to modify the Werft config and restart the service when adding or modifying the cron expressions for a cron job.

As Werft only detects jobs in `.werft` I had to move the existing cron jobs to `.werft` - previously this wasn't an issue as the cron definitions for each job in the Werft config pointed to a specific file. But now we're relying on Werft detecting the jobs in order to see the cron expression, so the job specifications need to live directly in `.werft`.

I introduced the same naming convention as we have in gitpod-io/ops, which is `<team name>-<job-name>` and updated CODEOWNERS.

I deleted the `delete-preview-environments` job and instead changed the `delete-preview-environments-cron` job to run every hour. Werft never triggered `delete-preview-environments` when branches were deleted so we have never seen the job work. Cleaning preview environments every hour is good enough for now.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/1255

## How to test


Once merged we have to verify that Werft does indeed schedule the cron jobs. As the delete preview environments cron job now runs every hour I will use that one to verify that it gets triggered.

## Release Notes

```release-note
NONE
```

## Documentation

I have to updated our How-to guides on how to add cron jobs.
